pkg/planner/core/find_best_task.go

```diff
@@ -3108,6 +3161,14 @@ func getOriginalPhysicalIndexScan(ds *logicalop.DataSource, prop *property.Physi
 }
 
 func findBestTask4LogicalCTE(lp base.LogicalPlan, prop *property.PhysicalProperty, counter *base.PlanCounterTp, pop *optimizetrace.PhysicalOptimizeOp) (t base.Task, cntPlan int64, err error) {
+	// if prop is require an index join's probe side, check the inner pattern admission here.
+	if prop.IndexJoinProp != nil {
+		pass := admitIndexJoinInnerChildPattern(lp)
+		if !pass {
+			// even enforce hint can not work with this.
+			return base.InvalidTask, 0, nil
+		}
```

Comment:
Don't we just return `invalidTask` directly?

Reply:
make sense, we don't need to chheck the pattern, because findBestTask4xxx, the operator is determined. changed:https://github.com/pingcap/tidb/pull/60495/commits/20d62db15183c4a9246bc6efda417ea384df9d8f